### PR TITLE
feat: implement `MacosDisclosureButton`

### PR DESCRIPTION
**Pull request by Umar1312**
## Pre-launch Checklist

- [x] I have run `dartfmt` on all changed files 
- [x] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes 
- [x] I have added/updated relevant documentation 
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could

### CODE REVIEW

**Comment by GroovinChip** [comment]
Hi @Umar1312, thanks for opening this PR! 

Please address the failing check, and then I can review.

**Comment by Umar1312** [comment]
> Hi @Umar1312, thanks for opening this PR!
> 
> Please address the failing check, and then I can review.

Addressed. Please check. 

Thanks!

**Comment by GroovinChip** [comment]
@Umar1312 The Flutter analysis check is still failing, looks like your test file.

**Comment by GroovinChip** [comment]
Also, if you could please upload a gif of the widget in action, I'd really appreciate it. I don't generally have the time to pull PR branches and manually run the example to test the new code, so gifs are essential to my reviews.

**Comment by Umar1312** [comment]
Addressed. Also PFA the gif of widget in action. 

Thanks!

DisclosureButtonGIf

File: lib/src/buttons/disclosure_button.dart [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+              opacity: _opacityAnimation,
+              child: AnimatedBuilder(
+                animation: _opacityAnimation,
+                builder: (context, widget1) {
+                  return DecoratedBox(
+                    decoration: BoxDecoration(
+                      color: buttonHeldDown
+                          ? brightness == Brightness.dark
+                              ? const Color(0xff3C383C)
+                              : const Color(0xffE5E5E5)
```
Comment (GroovinChip) [inline lib/src/buttons/disclosure_button.dart]:
```suggestion
                              ? const MacosColor(0xff3C383C)
                              : const MacosColor(0xffE5E5E5)
```

**Review by GroovinChip — APPROVED** [review]
OK so I've done a more thorough review and there are some fundamental changes that should be made:

- The disclosure control that is used for showing and hiding information and functionality associated with a view or list of items is called a [Disclosure triangle](https://developer.apple.com/design/human-interface-guidelines/components/layout-and-organization/disclosure-controls#disclosure-triangles). The widget should be updated to reflect this.
- The disclosure control that is used for showing and hiding functionality associated with a specific control is called a [Disclosure button](https://developer.apple.com/design/human-interface-guidelines/components/layout-and-organization/disclosure-controls#disclosure-buttons). This one looks like what you've made, but the chevron points **down**.

I think for both of these modes, the widget should operate in a similar manner to Flutter's [ExpansionTile](https://api.flutter.dev/flutter/material/ExpansionTile-class.html), which can show/hide other widgets.

**Review by GroovinChip — CHANGES_REQUESTED** [review]
Thought I'd done this as part of my [previous comment](https://github.com/GroovinChip/macos_ui/pull/326#pullrequestreview-1267395621) but I guess not. Please see that and make the required changes. Thanks!

**Comment by Umar1312** [comment]
@GroovinChip Thanks for reviewing. I'm not quite sure on what changes I should make for this.

1. Going through the Apple's Docs, what we are trying to to implement is `NSBezelStyleRoundedDisclosure`. The actual  implementation of the hide/show part should be the responsibility of the user as is the case for Apple's own implementation.

2. If you still want to incorporate the show/hide functionality then we should probably not view it as a button widget, as it's scope has now changed considerably. 


Let me know what you think about this.

**Comment by GroovinChip** [comment]
@Umar1312 I'll think about this and get back to you, thanks.

**Comment by GroovinChip** [comment]
OK @Umar1312 we can keep things as they are. Before I review, please update your branch with the latest updates from `dev`.

**Comment by Umar1312** [comment]
@GroovinChip Thanks, updated.

**Comment by GroovinChip** [comment]
@Umar1312 I actually just overhauled `MacosScrollBar` which allows the return of the `ScrollController` to `ContentArea`. If you want I can land that first and you can revert 2fa9902, or we can land this one and I'll fix it in the PR to land the changes to `MacosScrollBar`. What would you like to do?

EDIT: Let's actually land this one first, don't worry about it :)

File: lib/src/buttons/disclosure_button.dart [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+  const MacosDisclosureButton({
+    super.key,
+    this.fillColor,
+    this.semanticLabel,
+    this.isPressed = false,
+    this.mouseCursor = SystemMouseCursors.basic,
+    this.onPressed,
+  });
+
+  final VoidCallback? onPressed;
```
Comment (GroovinChip) [inline lib/src/buttons/disclosure_button.dart:21]: Document this please

File: lib/src/buttons/disclosure_button.dart [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  /// The color to fill the space around the icon with.
+  final Color? fillColor;
+
+  /// The semantic label used by screen readers.
+  final String? semanticLabel;
+
+  /// The mouse cursor to use when hovering over this widget.
+  final MouseCursor? mouseCursor;
+
+  final bool isPressed;
```
Comment (GroovinChip) [inline lib/src/buttons/disclosure_button.dart:34]: Document this please

File: lib/src/buttons/disclosure_button.dart [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+  /// Whether the button is enabled or disabled. Buttons are disabled by default. To
+  /// enable a button, set its [onPressed] property to a non-null value.
```
Comment (GroovinChip) [inline lib/src/buttons/disclosure_button.dart]:
```suggestion
  /// Whether the button is enabled or disabled. Buttons are disabled by default. 
  ///
  /// To enable a button, set its [onPressed] property to a non-null value.
```

**Review by GroovinChip — CHANGES_REQUESTED** [review]
Just some minor documentation requests :)

**Comment by Umar1312** [comment]
@GroovinChip Updated.

**Review by GroovinChip — APPROVED** [review]
LGTM 🎉

**Pull request merged**